KBH/12Day/실패율.js

```diff
@@ -0,0 +1,35 @@
+function solution(N, stages) {
+    const answer = [];
+    const copyedStages = [...stages];
+    //실패한 스테이지 개수 확인
+    const failedStage = copyedStages.reduce((acc, cur) => { acc[cur] = (acc[cur] || 0 ) + 1; 
+        return acc;
+    }, {});
+    
+    //다음 스테이지에 올라간 유저수
+    let stageLength = stages.length;
+    //실패율 변수 선언
+    let failedRate = [];
+
+
+    copyedStages.sort();
+
+    for (let i = 1; i <= N; i++) {
+        //해당 객체로 접근해서 사용자의 수를 찾음.
+        const failedCount = failedStage[i] || 0;
+        //실패율 계산
+        failedRate.push({stage : i, rate : failedCount / stageLength});
+        //실패율 계산 후 다음 스테이지에 올라온 유저를 빼기 위해 변수 선언
+        stageLength -= failedCount;
+
+    }
+    //sort 함수를 이용해서 rate의 값을 기준으로 내림차순 진행
+    failedRate.sort((a, b) => b.rate - a.rate);
+
+    //for문을 이용해서 각 인덱스마다 stage요소로 접근해 answer 배열에 저장
+    for (let j = 0; j < failedRate.length; j++) {
+        answer.push(failedRate[j].stage);
+    }
```

Comment:
```suggestion
    for (let i = 0; i < failedRate.length; i++) {
        answer.push(failedRate[i].stage);
    }
```
일반적으로 반복문은 i 2중 반복문일때는 i, j로 작성됩니다 위 코드의 경우에는 위쪽의 반복문과 스코프가 다르기 때문에 굳이 j로 선언해 줄 필요가 없어 보이네요 👍 의도한 것 이라면 어쩔 수 없지만
일반적인 코드 컨벤션을 따라가는것이 좋아보여요!😊